### ISSUE TITLE
Leaves with no training data affect parents

### ISSUE THREAD

**Issue by hacktuarial** [globality-corp/sklearn-hierarchical-classification#31]
# Description
Given a graph with an empty leaf (no training data), I expected that `HierarchicalClassifier` would essentially drop it, and use its parent node instead. However, it seems that the parent node is being dropped as well. 

# Code

```python
import itertools 
from collections import Counter

import networkx as nx
import sklearn_hierarchical_classification
from sklearn_hierarchical_classification.classifier import HierarchicalClassifier
from sklearn.linear_model import LogisticRegression
import numpy as np

# simple clothing hierarchy
G = nx.DiGraph()
ROOT = "root"
G.add_nodes_from([ROOT, "Mens", ])
G.add_edge(ROOT, "Mens")
G.add_edge("Mens", "Shirts")
G.add_edge("Mens", "Bottoms")
G.add_edge("Mens", "Jackets")
G.add_edge("Mens", "Swim")
G.nodes()

# generate some data
np.random.choice(1235)
categories = G.nodes()
y = np.random.choice(categories, size=50)
# make sure all nodes were selected
assert set(y) == set(categories)
X = np.random.normal(size=(len(y), 10))
base_estimator = LogisticRegression(solver="lbfgs", max_iter=1_000, multi_class="multinomial")

# this works fine
hc = HierarchicalClassifier(base_estimator,
                            class_hierarchy=G,
                            algorithm="lcpn",
                            root=ROOT)
hc.fit(X, y)
for node in G.nodes:
    if G.out_degree(node) > 0:
        preds = hc.graph_.nodes()[node]["classifier"].predict_proba(X[0:1, :])
        assert preds.shape[1] == G.out_degree[node]


# now, modify G so it has an empty leaf
G.add_edge("Bottoms", "Pants")
assert not any(yi == "Pants" for yi in y)
hc2 = HierarchicalClassifier(base_estimator,
                            class_hierarchy=G,
                            algorithm="lcpn",
                            root=ROOT)
hc2.fit(X, y) # issues expected warning, "...will terminate at node Bottoms"

# this fails
assert "Bottoms" in hc2.graph_.nodes()["Mens"]["classifier"].classes_, hc2.graph_.nodes()["Mens"]["classifier"].classes_
```

# Expected Results
I expect the classifier in the "Mens" node to have learned 4 classes, corresponding to the DiGraph: Shirts, Bottoms, Jackets, and Swim. 

# Actual Results
The classifier in the "Mens" node dropped the label for "Bottoms" 

# Versions
sklearn-hierarchical-classification == 1.1.0
Output of `import sklearn; sklearn.show_versions()` follows:

System:
    python: 3.6.7 (default, May 13 2019, 16:14:45)  [GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)]
executable: /Users/tsweetser/.pyenv/versions/3.6.7/envs/jupyter/bin/python
   machine: Darwin-18.7.0-x86_64-i386-64bit

Python deps:
       pip: 19.2.3
setuptools: 41.0.1
   sklearn: 0.21.3
     numpy: 1.17.2
     scipy: 1.3.1
    Cython: 0.29.6
    pandas: 0.24.2

**Comment by adamhadani** [comment]
thank you for the very detailed report @hishacktuarial ! helped a lot in debugging and solving this.

Solved via #32 - look out for a new release `1.3.0` which i'll make later today after merging.
